pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -6,44 +6,127 @@ import (
 	"reflect"
 	"time"
 
-	inventory "github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/inventory"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/meta"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/openshift"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/builder"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
 
 	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/logs"
+	kogitocli "github.com/kiegroup/kogito-cloud-operator/pkg/client"
 	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/shared"
 	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/status"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/logger"
 	oappsv1 "github.com/openshift/api/apps/v1"
 	obuildv1 "github.com/openshift/api/build/v1"
+	oimagev1 "github.com/openshift/api/image/v1"
 	oroutev1 "github.com/openshift/api/route/v1"
+	routev1 "github.com/openshift/api/route/v1"
 	buildv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
 	imagev1 "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	cachev1 "sigs.k8s.io/controller-runtime/pkg/cache"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = logs.GetLogger("controller_kogitoapp")
+var log = logger.GetLogger("controller_kogitoapp")
+
+// Add creates a new KogitoApp Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	imageClient, err := imagev1.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Errorf("Error getting image client: %v", err)
+		return &ReconcileKogitoApp{}
```

Comment:
Not sure how this bubbles up, but you are expected to return the error here, which would presumably be a second return argument. The caller would check it and decide if to proceed.

Reply:
This function is generated by the SDK itself and the interface doesn't have a way to return an error in its signature. We're using the image and build clients for specific OpenShift use cases for this version (should be removed in later versions). Unfortunately those clients builders return an error and there's no other way to handle it (despite logging). Nor the `reconciler` or the `manager` has an interface to set the error.

I completely agree with you, so I'm going to `panic` in this method so the operator won't come up and the user should see in the logs what's happened.